src/main/java/mate/academy/internetshop/dao/jdbc/OrderDaoJdbcImpl.java

```diff
@@ -0,0 +1,149 @@
+package mate.academy.internetshop.dao.jdbc;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.internetshop.dao.OrderDao;
+import mate.academy.internetshop.exceptions.DataProcessingException;
+import mate.academy.internetshop.lib.Dao;
+import mate.academy.internetshop.model.Order;
+import mate.academy.internetshop.model.Product;
+import mate.academy.internetshop.util.ConnectionUtil;
+
+@Dao
+public class OrderDaoJdbcImpl implements OrderDao {
+    @Override
+    public Order create(Order element) {
+        String query = "INSERT INTO orders (user_id) VALUES (?)";
+        String query2 = "INSERT INTO orders_products (product_id, order_id) VALUES (?, ?)";
+        Long key;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement =
+                    connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS);
+            statement.setLong(1, element.getUserId());
+            statement.execute();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            generatedKeys.next();
+            key = generatedKeys.getLong(1);
+            for (Product product : element.getProducts()) {
+                statement = connection.prepareStatement(query2);
+                statement.setLong(1, product.getId());
+                statement.setLong(2, key);
+                statement.execute();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create order in DataBase");
+        }
+        return get(key).get();
+    }
+
+    @Override
+    public Optional<Order> get(Long id) {
+        String query = "SELECT * FROM orders "
+                + "LEFT JOIN orders_products ON orders.order_id = orders_products.order_id "
+                + "LEFT JOIN products ON orders_products.product_id = products.product_id "
```

Comment:
do you know the difference between left join and join?And which of them is better to use in this case?Explain in comment below, please

Reply:
Yeah, maybe its better to use inner join because we will have matches in any case, left join we use only when we want to take everything from left table even if we have null in the right table